internal/controller/profiles/common/knative_eventing.go

```diff
@@ -67,6 +70,9 @@ func (k knativeObjectManager) Ensure(ctx context.Context, workflow *operatorapi.
 			return objs, err
 		} else if sinkBinding != nil {
 			objs = append(objs, sinkBinding)
+			if err := k.cleanupOutdatedRevisions(ctx, workflow, sinkBinding); err != nil {
```

Comment:
We have a post-hook implementation that we can use to garbage collect it. WDYT? Also, `sinkBinding` is not used anywhere.

https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/main/internal/controller/profiles/profile.go#L74

So, in the reconciliation states, we can have a GC calling `cleanupOutdatedRevisions`.

Reply:
> We have a post-hook implementation that we can use to garbage collect it. WDYT? Also, `sinkBinding` is not used anywhere.
> 
> https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/main/internal/controller/profiles/profile.go#L74
> 
> So, in the reconciliation states, we can have a GC calling `cleanupOutdatedRevisions`.

Sounds good. PR updated.